includes/User.php

```diff
@@ -592,6 +592,40 @@ public static function whoIs( $id ) {
 	}
 
 	/**
+	 *
+	 * @param $ids Array User IDs
+	 * @return Array User ID to User name mapping
+	 */
+	public static function whoAre( Array $ids, $source = DB_SLAVE ): Array {
+		global $wgExternalSharedDB;
+		$sdb = wfGetDB( $source, [], $wgExternalSharedDB );
+		$res = $sdb->select(
+			'`user`',
+			[ 'DISTINCT user_id', 'user_name' ],
+			[ 'user_id' => $ids ],
```

Comment:
Shouldn't we call `array_unique` on `$ids` instead of relying on MySQL doing `DISTINCT`?

Reply:
Why?